src/pybind/mgr/nfs/export.py

```diff
@@ -482,46 +488,44 @@ def get_export_by_pseudo(
-                # okay, not JSON.  is it an EXPORT block?
+            aeresults = AppliedExportResults()
+            for export in self._read_export_config(cluster_id, export_config):
                 try:
-                    blocks = GaneshaConfParser(export_config).parse()
-                    exports = [
-                        Export.from_export_block(block, cluster_id)
-                        for block in blocks
-                    ]
-                    j = [export.to_dict() for export in exports]
+                    chg = self._apply_export(cluster_id, export)
                 except Exception as ex:
-                    raise NFSInvalidOperation(f"Input must be JSON or a ganesha EXPORT block: {ex}")
-
-            # check export type
-            if isinstance(j, list):
-                ret, out, err = (0, '', '')
-                for export in j:
-                    try:
-                        r, o, e = self._apply_export(cluster_id, export)
-                    except Exception as ex:
-                        r, o, e = exception_handler(ex, f'Failed to apply export: {ex}')
-                        if r:
-                            ret = r
-                    if o:
-                        out += o + '\n'
-                    if e:
-                        err += e + '\n'
-                return ret, out, err
-            else:
-                r, o, e = self._apply_export(cluster_id, j)
-                return r, o, e
+                    msg = f'Failed to apply export: {ex}'
+                    log.exception(msg)
+                    chg = {"state": "error", "msg": msg}
+                aeresults.append(chg)
+            return aeresults
         except NotImplementedError:
-            return 0, " Manual Restart of NFS PODS required for successful update of exports", ""
+            raise ManualRestartRequired("Changes applied")
         except Exception as e:
-            return exception_handler(e, f'Failed to update export: {e}')
+            log.exception(f'Failed to update export: {e}')
+            raise ErrorResponse.wrap(e)
+
+    def _read_export_config(self, cluster_id: str, export_config: str) -> List[Dict]:
+        if not export_config:
+            raise NFSInvalidOperation("Empty Config!!")
+        try:
+            j = json.loads(export_config)
+        except ValueError:
+            # okay, not JSON.  is it an EXPORT block?
+            try:
+                blocks = GaneshaConfParser(export_config).parse()
+                exports = [
+                    Export.from_export_block(block, cluster_id)
+                    for block in blocks
+                ]
+                j = [export.to_dict() for export in exports]
+            except Exception as ex:
+                raise NFSInvalidOperation(f"Input must be JSON or a ganesha EXPORT block: {ex}")
+        # check export type - always return a list
+        if isinstance(j, list):
+            return j
+        return [j]
```

Comment:
equivalent to: `return j if isinstance(j, list) else [j]`

Reply:
sure but I like the multi line version, I think it is easier to see the difference of `j` vs. `[j]` . I think I will also add comments when I update the code.

Reply:
comments added